mlflow/sklearn/__init__.py

```diff
@@ -514,9 +524,10 @@ def __init__(self, sklearn_model):
 
         # Patch the model with custom predict functions that can be specified
         # via `pyfunc_predict_fn` argument when saving or logging.
-        for predict_fn in self._SUPPORTED_CUSTOM_PREDICT_FN:
-            if fn := getattr(self.sklearn_model, predict_fn, None):
-                setattr(self, predict_fn, fn)
+        for predict_fn_name in self._SUPPORTED_CUSTOM_PREDICT_FN:
+            if fn := getattr(self.sklearn_model, predict_fn_name, None):
+                predict_fn = _gen_sklearn_model_predict_fn(fn, predict_fn_name)
+                setattr(self, predict_fn_name, predict_fn)
```

Comment:
we just might want to define each method for readbility.

```
def predict_proba(...):
    ...

def predict_log_proba(...):
    ...
```

Reply:
do we need this ? they all have the same pattern.

Reply:
Never mind this comment. unrelated to this PR. I'm just not a big fan of `getattr` or `setattr`.